helix-core/src/text_size/range.rs

```diff
@@ -0,0 +1,400 @@
+use core::fmt;
+use std::{
+    cmp::{self, Ordering},
+    ops::{self, Add, Sub},
+};
+
+use super::size::TextSize;
+
+/// A range in text, represented as a pair of [`TextSize`][struct@TextSize].
+///
+/// It is a logic error for `start` to be greater than `end`.
+#[derive(Default, Copy, Clone, Eq, PartialEq, Hash)]
+pub struct TextRange {
```

Comment:
Isn't this basically just `SelectionRange`?

Reply:
`SelectionRange` can be flipped